index.js

```diff
@@ -117,21 +129,39 @@ var processor = postcss(plugins);
 // hook for dynamically updating the list of watched files
 global.watchCSS = function() {};
 if (argv.watch) {
-  var watchedFiles = inputFiles;
+  global.watchCSS = fsWatcher(inputFiles);
+}
+
+async.forEach(inputFiles, compile, onError);
+
+function fsWatcher(entryPoints) {
+  var watchedFiles = entryPoints;
+  var index = {}; // source files by entry point
+
   var watcher = require('chokidar').watch(watchedFiles);
-  watcher.on('change', function() { // TODO: support for "add", "unlink" etc.?
-    async.forEach(inputFiles, compile, function(err) {
+  // recompile if any watched file is modified
+  // TODO: only recompile relevant entry point
```

Comment:
I didn't bother to look into implementing this potential optimization. It would require a reverse lookup of the respective entry point via the file that's been changed.

Perhaps PostCSS already prevents recompiling if nothing has changed?